nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2307,11 +2315,75 @@ async fn ip_pool_service_range_remove(
 
 // Disks
 
+#[endpoint {
+    method = GET,
+    path = "/v1/disks",
+    tags = ["disks"],
+}]
```

Comment:
> **Warning**
> I've taken liberties here. This code warrants extra attention. 

Previously we had
- `GET /.../projects/{project}/disks`
- `GET /.../instances/{instance}/disks`


Essentially this allows listing all disks in the first case and only disks attached to a particular instance in the second case. I've collapsed this down into a single endpoint. 

- `GET /v1/disks?project={project_id}`
- `GET /v1/disks?instance={instance_id}`

If you provide the instance selector then it filters down to disks attached to the instance. Otherwise it gives you all disks in the projects. That makes this endpoint have slightly different requirement semantics on its selectors. In general the selector for the parent resource (`project` in this case) is _required_ for list endpoints. Because the selector here can be _either_ an instance or project (and their subsequent paths) all the selectors are optional. 

Overall I think the _usage_ of this is simplified and it gives one place to go to list disks. Let me know what you think.

Reply:
In line with https://github.com/oxidecomputer/omicron/pull/2008#discussion_r1053966938

Reply:
I'm not sure what you mean here.

Reply:
I can see the appeal.  I'm not sure I grok the implications but it sounds reasonable!

Reply:
Ha, well, I did an about face. This no longer applies. The endpoint is now `/v1/instances/{instance}/disks` which maps to what we had previously.

Reply:
I guess I'm fine with either. You had me convinced (and enthusiastic!) about a single endpoint with multiple search criteria, but I'm fine with the "about face". What changed your mind?

Reply:
A few reasons:
- Keeping `/disks` on the end of instances is consistent with the current API so this is the minimal viable change. This is the main one. 
- A consequence of being able to select on `instance` or `project` is that they both had to be optional. That just means there's an extra failure mode here. There's also a bit more to document.

I will say that I like the single endpoint better (what I had) than I like having two endpoints returning the same type of things at different levels of the API (what's there now and before this PR). If we already have a place that we're interacting with disks associated with an instance though, it probably just makes sense to add the listing there. 

Overall I would be good to move back to the single `disk` list endpoint, but we'd need to do something with the attach/detach APIs. I don't think we'd want to have `instances/{}/disks/(attach|detach)` because the `/disks` part would be hanging with no valid methods on it. If we dropped `/disks` like @karencfv original suggestion then `/instances/{}/(attach|detach)` may be too vague. What are you attaching or detaching? Lastly I just wasn't a fan of chaining names together where we don't have to `/disks-attach`. So, that's why I ended up just going back to the original.